pkg/model/api/v1/secret/tls_config.go

```diff
@@ -26,8 +26,11 @@ type PublicTLSConfig struct {
 	InsecureSkipVerify bool   `yaml:"insecureSkipVerify" json:"insecureSkipVerify"`
 }
 
-func NewPublicTLSConfig(t TLSConfig) PublicTLSConfig {
-	return PublicTLSConfig{
+func NewPublicTLSConfig(t *TLSConfig) *PublicTLSConfig {
```

Comment:
Use pointer like others fields in order to not create struct if the copied secret doesn't have TLS config. 
Previously empty tls config was copied to {InsecureSkipVerify: false}

Reply:
you need to complete this condition https://github.com/perses/perses/blob/main/internal/api/core/middleware/proxy.go#L400 otherwise we will have a panic at runtime